OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -221,14 +221,16 @@ WAngle DesiredLocalFacing
 				if (desiredDirection == WVec.Zero)
 					return LocalOrientation.Yaw;
 
+				if (facing == null)
+					return desiredDirection.Yaw;
+
 				// PERF: If the turret rotation axis is vertical we can directly take the difference in facing/yaw
-				var o = facing != null ? facing.Orientation : (WRot?)null;
-				if (o == null || (o.Value.Pitch == WAngle.Zero && o.Value.Roll == WAngle.Zero))
-					return o.HasValue ? desiredDirection.Yaw - o.Value.Yaw : desiredDirection.Yaw;
+				if (facing.Orientation.Pitch == WAngle.Zero && facing.Orientation.Roll == WAngle.Zero)
```

Comment:
We can't guarantee that `facing.Orientation` won't do unnecessary calculations (or even return the same result) for each of the four uses, so can we please keep a `var o = facing.Orientation`?

Reply:
"or even return the same result" it has to be the same result since we're not ticking inbetween and the getter shouldn't change the orientation, otherwise something else is really bugged. I'll restore it for caching.